synapse/crypto/context_factory.py

```diff
@@ -191,9 +191,9 @@ def _context_info_cb(ssl_connection, where, ret):
         # ... we further assume that SSLClientConnectionCreator has set the
         # '_synapse_tls_verifier' attribute to a ConnectionVerifier object.
         tls_protocol._synapse_tls_verifier.verify_context_info_cb(ssl_connection, where)
-    except:  # noqa: E722, taken from the twisted implementation
+    except BaseException as e:  # taken from the twisted implementation
```

Comment:
I'm not entirely sure if this is correct, but "taken from the twisted implementation" tells me that twisted (more or less) explicitly intended to also capture `BaseException` as well.

Reply:
I think we prefer *not* passing the error to `Failure` since that doesn't magic to pull out the "real" failure (or a failure with a better stack trace or something).

Reply:
Oh huh, I'll change it then 👍